pkg/nvlib/device/identifier_test.go

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) NVIDIA CORPORATION.  All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package device
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestIsGpuIndex(t *testing.T) {
+	testCases := []struct {
+		id       string
+		expected bool
+	}{
+		{"", false},
+		{"0", true},
```

Comment:
I think we require non-negative indices. Should we add a test for `"-1"`?

Reply:
Good call-out. I have added a test case for negative integers.